lms/djangoapps/grades/api/v1/views.py

```diff
@@ -0,0 +1,202 @@
+            raise CourseEnrollment.DoesNotExist
+
+        course_grade = CourseGradeFactory().read(grade_user, course_key=course_key)
+        return Response([self._make_grade_response(grade_user, course_key, course_grade)])
+
+    def _get_user_grades(self, course_key):
+        """
+        Get paginated grades for users in a course.
+        Args:
+            course_key (CourseLocator): The course to retrieve user grades for.
+
+        Returns:
+            A serializable list of grade responses
+        """
+        enrollments_in_course = enrollment_data.get_user_enrollments(course_key)
+
+        paged_enrollments = self.paginator.paginate_queryset(
+            enrollments_in_course, self.request, view=self
+        )
+        users = (enrollment.user for enrollment in paged_enrollments)
+        grades = CourseGradeFactory().iter(users, course_key=course_key)
+
+        grade_responses = []
+        for user, course_grade, exc in grades:
+            if not exc:
+                grade_responses.append(self._make_grade_response(user, course_key, course_grade))
+
+        return Response(grade_responses)
+
+    def _make_grade_response(self, user, course_key, course_grade):
+        """
+        Serialize a single grade to dict to use in Responses
+        """
+        return {
+            'username': user.username,
+            'email': user.email,
+            'course_id': str(course_key),
+            'passed': course_grade.passed,
+            'percent': course_grade.percent,
+            'letter_grade': course_grade.letter_grade,
+        }
+
+    def perform_authentication(self, request):
+        """
+        Ensures that the user is authenticated (e.g. not an AnonymousUser).
+        """
+        super(GradeViewMixin, self).perform_authentication(request)
+        if request.user.is_anonymous():
+            raise AuthenticationFailed
+
+
+class CourseGradesView(GradeViewMixin, GenericAPIView):
+    """
+    **Use Case**
+        * Get course grades of all users who are enrolled in a course.
+        The currently logged-in user may request all enrolled user's grades information
+        if they are allowed.
+    **Example Request**
+        GET /api/grades/v1/courses/{course_id}/                              - Get grades for all users in course
+        GET /api/grades/v1/courses/{course_id}/?username={username}          - Get grades for specific user in course
+        GET /api/grades/v1/courses/?course_id={course_id}                    - Get grades for all users in course
+        GET /api/grades/v1/courses/?course_id={course_id}&username={username}- Get grades for specific user in course
+    **GET Parameters**
+        A GET request may include the following parameters.
+        * course_id: (required) A string representation of a Course ID.
+        * username:  (optional) A string representation of a user's username.
+    **GET Response Values**
+        If the request f    or information about the course grade
```

Comment:
@nasthagiri so you want me to manually squash commits and then rebase?
I've just been pulling the latest master changes each time I make new commits. 

Generally we use the squash PR functionality inside github when merging pull requests to achieve this effect. At this point it's a bit non-trivial to squash all these commits and then rebase without using a different branch.

Let me know if you have any guidance.

Reply:
I see. Here, we generally do this via the command line:

```
# to squash all 21 commits
git rebase -i HEAD~21
# in the editor, replace all "pick" with "s" - except for the 1st commit

# to rebase with master
git fetch; git rebase origin/master

# to push back changes (you need a force-push since you've rebased)
git push -f
```

@nedbat Do you have alternative advice for @kabirkhan?

Reply:
@nasthagiri @nedbat the end effect should be the same either way: a single squashed commit merged to master.

Reply:
@kabirkhan I would be the one merging this PR and that option is not available to me in this repo:
![screen shot 2018-05-02 at 5 17 50 pm](https://user-images.githubusercontent.com/6772826/39549691-c9d9e2c4-4e2c-11e8-8956-3e2eba5849ca.png)

Reply:
@nasthagiri I believe I've done this properly.

Reply:
Ok. great!  As soon as the build is green, I'll merge.